pkgs/by-name/ca/cantata/package.nix

```diff
@@ -0,0 +1,201 @@
+      enable = withCdda;
+      pkgs = [ cdparanoia ];
+    }
+    {
+      names = [ "DEVICES_SUPPORT" ];
+      enable = withDevices;
+      pkgs = [ ];
+    }
+    {
+      names = [ "DYNAMIC" ];
+      enable = withDynamic;
+      pkgs = [ ];
+    }
+    {
+      names = [
+        "FFMPEG"
+        "MPG123"
+        "SPEEXDSP"
+      ];
+      enable = withReplaygain;
+      pkgs = [
+        ffmpeg
+        speex
+        mpg123
+      ];
+    }
+    {
+      names = [ "HTTPS_SUPPORT" ];
+      enable = true;
+      pkgs = [ ];
+    }
+    {
+      names = [ "HTTP_SERVER" ];
+      enable = withHttpServer;
+      pkgs = [ ];
+    }
+    {
+      names = [ "HTTP_STREAM_PLAYBACK" ];
+      enable = withHttpStream;
+      pkgs = [ qt5.qtmultimedia ];
+    }
+    {
+      names = [ "LAME" ];
+      enable = withLame;
+      pkgs = [ lame ];
+    }
+    {
+      names = [ "LIBVLC" ];
+      enable = withLibVlc;
+      pkgs = [ libvlc ];
+    }
+    {
+      names = [ "MTP" ];
+      enable = withMtp;
+      pkgs = [ libmtp ];
+    }
+    {
+      names = [ "MUSICBRAINZ" ];
+      enable = withMusicbrainz;
+      pkgs = [ libmusicbrainz5 ];
+    }
+    {
+      names = [ "ONLINE_SERVICES" ];
+      enable = withOnlineServices;
+      pkgs = [ ];
+    }
+    {
+      names = [ "STREAMS" ];
+      enable = withStreams;
+      pkgs = [ ];
+    }
+    {
+      names = [
+        "TAGLIB"
+        "TAGLIB_EXTRAS"
+      ];
+      enable = withTaglib;
+      pkgs = [
+        taglib
+        taglib_extras
+      ];
+    }
+    {
+      names = [ "UDISKS2" ];
+      enable = withUdisks;
+      pkgs = [ udisks2 ];
+    }
+  ];
+
+in
+stdenv.mkDerivation rec {
+  pname = "cantata";
+  version = "2.5.0";
+
+  src = fetchFromGitHub {
+    owner = "CDrummond";
+    repo = "cantata";
+    rev = "v${version}";
+    sha256 = "sha256-UaZEKZvCA50WsdQSSJQQ11KTK6rM4ouCHDX7pn3NlQw=";
+  };
+
+  patches = [
+    # Cantata wants to check if perl is in the PATH at runtime, but we
+    # patchShebangs the playlists scripts, making that unnecessary (perl will
+    # always be available because it's a dependency)
+    ./dont-check-for-perl-in-PATH.diff
+  ];
+
+  postPatch = ''
+    patchShebangs playlists
+  '';
+
+  buildInputs = [
+    qt5.qtbase
+    qt5.qtsvg
+    (perl.withPackages (ppkgs: with ppkgs; [ URI ]))
+  ] ++ lib.flatten (builtins.catAttrs "pkgs" (builtins.filter (e: e.enable) options));
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    qt5.qttools
+    qt5.wrapQtAppsHook
+  ];
+
+  cmakeFlags = lib.flatten (map (e: map (f: fstat e.enable f) e.names) options);
+
+  meta = {
+    description = "Graphical client for MPD";
+    mainProgram = "cantata";
+    homepage = "https://github.com/cdrummond/cantata";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ peterhoeg ];
+    # Technically, Cantata should run on Darwin/Windows so if someone wants to
+    # bother figuring that one out, be my guest.
```

Comment:
If this is actually a cross platform application, we should mark the platforms correctly and then set the ones it doesn't currently compile/run for as badPlatforms.

Reply:
Oh, I was saying we could just mark it badPlatform right away so the maintainer doesn't have to worry about maintaining another platform but to properly represent it's status as a cross platform application.